android/app/src/main/kotlin/coop/polypoly/polypod/FeatureCardAdapter.kt

```diff
@@ -55,13 +60,80 @@ class FeatureCardAdapter(
         view.setOnClickListener {
             // FIXME - navigation assumes we're coming from FirstFragment,
             // which might not necessary be true
-            val action =
-                FeatureListFragmentDirections
-                    .actionFeatureListFragmentToFeatureFragment(
-                        feature.name,
-                        feature.fileName
-                    )
-            findNavController(originatingFragment).navigate(action)
+            authorize {
+                val action =
+                    FeatureListFragmentDirections
+                        .actionFeatureListFragmentToFeatureFragment(
+                            feature.name,
+                            feature.fileName
+                        )
+                findNavController(originatingFragment).navigate(action)
+            }
+        }
+    }
+
+    fun biometricsUnavailable(): Boolean {
+        originatingFragment.context?.let {
+            val biometricManager = BiometricManager.from(it)
+            if (biometricManager.canAuthenticate(
+                    OnboardingActivity.desiredLockScreenType
+                ) == BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED
+            ) {
+                return true
+            }
+        }
+        return false
+    }
+
+    fun authorize(successfulAuth: (() -> Unit)) {
```

Comment:
Not entirely comfortable with the "successful" in the name. I guess it implies that it either authentifies, or raises an exception, but calling it successful before actually performing the thing is a bit jarring...

Reply:
This is only when authentication is succesful. Otherwise this is a noop.

Reply:
OK, let me nitpick a bit on this.
1. Are we talking about authentication or authorization? If we're doing the latter, maybe what should change is the name of the function.
2. The argument is a callback, and maybe some part of its name should reflect that. I can think about "cbOnSuccesfulAuth", for instance.
3. You're calling it also as a fallback in case there's no biometric authentication. So it's automatically successful if there's no biometric thing?